cluster-api/src/main/java/io/aiven/klaw/clusterapi/services/SchemaService.java

```diff
@@ -0,0 +1,208 @@
+      log.info("Schema versions " + responseList);
+      return responseList.getBody();
+    } catch (Exception e) {
+      log.error("Error in getting versions " + e.getMessage());
+      return new ArrayList<>();
+    }
+  }
+
+  private String getSchemaCompatibility(
+      String environmentVal, String topicName, KafkaSupportedProtocol protocol) {
+    try {
+      log.info("Into getSchema compatibility {} {}", topicName, environmentVal);
+      if (environmentVal == null) {
+        return null;
+      }
+
+      String suffixUrl = environmentVal + "/config/" + topicName + "-value";
+      Pair<String, RestTemplate> reqDetails =
+          clusterApiUtils.getRequestDetails(suffixUrl, protocol, KafkaClustersType.SCHEMA_REGISTRY);
+
+      Map<String, String> params = new HashMap<>();
+
+      ResponseEntity<HashMap> responseList =
+          reqDetails.getRight().getForEntity(reqDetails.getLeft(), HashMap.class, params);
+      log.info("Schema compatibility " + responseList);
+      return (String) responseList.getBody().get("compatibilityLevel");
+    } catch (Exception e) {
+      log.error("Error in getting schema compatibility " + e.getMessage());
+      return "NOT SET";
+    }
+  }
+
+  private boolean setSchemaCompatibility(
+      String environmentVal, String topicName, boolean isForce, KafkaSupportedProtocol protocol) {
+    try {
+      log.info("Into setSchema compatibility {} {}", topicName, environmentVal);
+      if (environmentVal == null) {
+        return false;
+      }
+
+      String suffixUrl = environmentVal + "/config/" + topicName + "-value";
+      Pair<String, RestTemplate> reqDetails =
+          clusterApiUtils.getRequestDetails(suffixUrl, protocol, KafkaClustersType.SCHEMA_REGISTRY);
+
+      Map<String, String> params = new HashMap<>();
+      if (isForce) {
+        params.put("compatibility", "NONE");
+      } else params.put("compatibility", defaultSchemaCompatibility);
+
+      HttpHeaders headers = new HttpHeaders();
+      headers.set("Content-Type", SCHEMA_REGISTRY_CONTENT_TYPE);
+      HttpEntity<Map<String, String>> request = new HttpEntity<>(params, headers);
+      reqDetails.getRight().put(reqDetails.getLeft(), request, String.class);
+      return true;
+    } catch (Exception e) {
+      log.error("Error in setting schema compatibility " + e.getMessage());
+      return false;
+    }
+  }
+
+  protected ClusterStatus getSchemaRegistryStatus(
+      String environmentVal, KafkaSupportedProtocol protocol) {
+
+    String suffixUrl = environmentVal + "/subjects";
+    Pair<String, RestTemplate> reqDetails =
+        clusterApiUtils.getRequestDetails(suffixUrl, protocol, KafkaClustersType.SCHEMA_REGISTRY);
+
+    Map<String, String> params = new HashMap<>();
+
+    try {
+      reqDetails.getRight().getForEntity(reqDetails.getLeft(), Object.class, params);
+      return ClusterStatus.ONLINE;
+    } catch (RestClientException e) {
+      e.printStackTrace();
```

Comment:
```suggestion
      log.error("Exception:", e);
```
it's better to log into  loggers rather than system out